qt/widgets/common/src/FunctionModel.cpp

```diff
@@ -44,40 +44,50 @@ void FunctionModel::setFunction(IFunction_sptr fun) {
 }
 
 IFunction_sptr FunctionModel::getFitFunction() const {
-  if (!m_function) {
+  if (!m_function)
     return m_function;
+
+  auto const numberOfFunctions = m_function->nFunctions();
+
+  if (numberOfFunctions > 1) {
+    if (m_currentDomainIndex < numberOfFunctions)
+      return getFitFunctionWithGlobals(m_currentDomainIndex);
+    return getFitFunctionWithGlobals(0);
+
+  } else if (numberOfFunctions == 1) {
+    auto const function = m_function->getFunction(0);
+    auto const composite =
+        std::dynamic_pointer_cast<CompositeFunction>(function);
+
+    if (composite && composite->nFunctions() == 1)
+      return composite->getFunction(0);
+    return function;
   }
-  auto const nf = m_function->nFunctions();
-  if (nf > 1) {
-    auto fun =
-        std::dynamic_pointer_cast<MultiDomainFunction>(m_function->clone());
-
-    auto const singleFun = m_function->getFunction(0);
-    for (auto par = m_globalParameterNames.begin();
-         par != m_globalParameterNames.end();) {
-      if (singleFun->hasParameter(par->toStdString())) {
-        QStringList ties;
-        for (size_t i = 1; i < nf; ++i) {
-          ties << "f" + QString::number(i) + "." + *par;
-        }
-        ties << "f0." + *par;
-        fun->addTies(ties.join("=").toStdString());
-        ++par;
-      } else {
-        par = m_globalParameterNames.erase(par);
-      }
-    }
-    return fun;
-  }
-  if (nf == 1) {
-    auto fun = m_function->getFunction(0);
-    auto compFun = std::dynamic_pointer_cast<CompositeFunction>(fun);
-    if (compFun && compFun->nFunctions() == 1) {
-      return compFun->getFunction(0);
+  return IFunction_sptr();
+}
+
+IFunction_sptr
+FunctionModel::getFitFunctionWithGlobals(std::size_t const &index) const {
+  auto function =
+      std::dynamic_pointer_cast<MultiDomainFunction>(m_function->clone());
+
+  auto const singleFun = m_function->getFunction(index);
+  for (auto paramIter = m_globalParameterNames.begin();
+       paramIter != m_globalParameterNames.end();) {
+    if (singleFun->hasParameter(paramIter->toStdString())) {
+      QStringList ties;
+      for (auto i = 0u; i < m_function->nFunctions(); ++i)
+        if (i != index)
+          ties << "f" + QString::number(i) + "." + *paramIter;
+
+      ties << "f" + QString::number(index) + "." + *paramIter;
```

Comment:
How come we have to add this last tie at the end, rather than constructing it within the loop? Is this to ensure the value for the parameter is equal to that defined in the currently selected function?

Reply:
That is a good question, and I didn't notice this as it was already here. I think the answer is yes. I assume the order in which the ties are applied matters if we want the global tie to be equal to the currently selected function.